zentral/contrib/osquery/api_views.py

```diff
@@ -416,6 +416,35 @@ def post(self, request, *args, **kwargs):
                         status=status.HTTP_201_CREATED)
 
 
+# Packs
+
+class PackFilter(filters.FilterSet):
+    configuration_id = filters.ModelChoiceFilter(field_name="configurationpack__configuration",
+                                                 queryset=Configuration.objects.all())
+    name = filters.CharFilter()
+
+
+class PackList(generics.ListCreateAPIView):
+    queryset = Pack.objects.all()
+    permission_classes = [DefaultDjangoModelPermissions]
+    serializer_class = PackSerializer
+    filter_backends = (filters.DjangoFilterBackend,)
+    filterset_class = PackFilter
+
+    def get(self, request, *args, **kwargs):
+        configuration_id = request.GET.get("configuration_id")
+        if configuration_id:
```

Comment:
Why do you need an exception here?

Reply:
@np5 
When the configuration exist, but it is not linked to any pack, the configuration_id filter will return an empty queryset, which is fine. 

But in the case, where the the configuration does not exist at all, we receive a default error that is not very useful. The message is: Select a valid choice. That choice is not one of the available choices. Id think we want to return a more explicit error message or at least always return an empty queryset, here I went for a default message.

That is why I created this check before calling the super method. 
So if the configuration does not exist and is not linked to any pack, we raise a ValidationError with a more clear message.

Reply:
I like the default behaviour better.

 * Valid configuration_id, no packs found → empty result
 * Invalid configuration_id, no configuration exists with the id → error saying that the choice is invalid, which is the case.

We already have the same error message when filtering by configuration_id on some other endpoints.